### PR TITLE
Feature/fix sendall

### PR DESCRIPTION
**Pull request by krlosMata**
This PR does the following:
- handle case when contract performs a `SENDALL` to itself

### CODE REVIEW

File: main/opcodes/create-terminate-context.zkasm [hunk 1]
```diff
@@ -960,20 +960,32 @@ opSENDALL:
     $ => A              :MLOAD(SP); [address => A]
```
Comment (ignasirv) [inline main/opcodes/create-terminate-context.zkasm:960]:
```suggestion
    $ => A              :MLOAD(SP), CALL(maskAddress); in: [A: address] out: [A: masked address] ; [address => A]
```

File: main/opcodes/create-terminate-context.zkasm [hunk 2]
```diff
@@ -996,17 +1008,20 @@ opSENDALL2:
     $ => A                  :MLOAD(storageAddr)
     0 => D
     $ => SR                 :SSTORE
+
+opSENDALLend:
     ; if origin ctx is 0, end tx
     $ => A                  :MLOAD(originCTX)
```
Comment (ignasirv) [inline main/opcodes/create-terminate-context.zkasm:1014]:
```suggestion
    $ => A                  :MLOAD(originCTX), JMPZ(handleGas)
```

File: main/opcodes/create-terminate-context.zkasm [hunk 3]
```diff
@@ -996,17 +1008,20 @@ opSENDALL2:
...
-    $                       :EQ,JMPC(handleGas)
+    A                       :JMPZ(handleGas)
+
     ; decrease CTX
     $ => B                  :MLOAD(gasRefund)
     $ => C                  :MLOAD(isCreate)
     1 => D
```
Comment (ignasirv) [inline main/opcodes/create-terminate-context.zkasm:1020]: WE could do this with one step less like this I guess:
....
$ => B                  :MLOAD(gasRefund)
1 => D
$ => C                  :MLOAD(isCreate), JMPZ(opSENDALLendContinue)
$ => D                  :MLOAD(createContractAddress)

opSENDALLendContinue:
....

File: main/opcodes/create-terminate-context.zkasm [hunk 4]
```diff
@@ -960,20 +960,32 @@ opSENDALL:
     $ => A              :MLOAD(SP); [address => A]
     ; Mask address to 20 bytes
                     :CALL(maskAddress); in: [A: address] out: [A: masked address]
-    ; check if storage address has balance to withdraw
+
+    ; store address receiver in [C, E]
     A => C
```
Comment (ignasirv) [inline main/opcodes/create-terminate-context.zkasm:965]: Can be optimized:
A => C, E

File: main/opcodes/create-terminate-context.zkasm (hunk 1 above)
Comment (krlosMata) [reply main/opcodes/create-terminate-context.zkasm:960]: added

File: main/opcodes/create-terminate-context.zkasm (hunk 2 above)
Comment (krlosMata) [reply main/opcodes/create-terminate-context.zkasm:1014]: added

File: main/opcodes/create-terminate-context.zkasm (hunk 3 above)
Comment (krlosMata) [reply main/opcodes/create-terminate-context.zkasm:1020]: added

File: main/opcodes/create-terminate-context.zkasm (hunk 4 above)
Comment (krlosMata) [reply main/opcodes/create-terminate-context.zkasm:965]: added

**Review by ignasirv — APPROVED** [review]
💯

**Pull request merged**